lib/rules/no-useless-lookaround-assertions.ts

```diff
@@ -0,0 +1,115 @@
+import type {
+    LookaheadAssertion,
+    LookaroundAssertion,
+    LookbehindAssertion,
+} from "regexpp/ast"
+import type { RegExpVisitor } from "regexpp/visitor"
+import type { RegExpContext } from "../utils"
+import { createRule, defineRegexpVisitor } from "../utils"
+
+export default createRule("no-useless-lookaround-assertions", {
+    meta: {
+        docs: {
+            description: "disallow useless nested lookaround assertions",
+            category: "Best Practices",
+            // TODO Switch to recommended in the major version.
+            // recommended: true,
+            recommended: false,
+        },
+        fixable: "code",
+        schema: [],
+        messages: {
+            unexpected: "This {{kind}} assertion is useless.",
+        },
+        type: "suggestion",
+    },
+    create(context) {
+        /**
+         * Create visitor
+         */
+        function createVisitor(
+            regexpContext: RegExpContext,
+        ): RegExpVisitor.Handlers {
+            return {
+                onAssertionEnter(aNode) {
+                    if (aNode.kind === "lookahead") {
+                        verifyForLookahead(regexpContext, aNode)
+                    } else if (aNode.kind === "lookbehind") {
+                        verifyForLookbehind(regexpContext, aNode)
+                    }
+                },
+            }
+        }
+
+        /** Verify for lookahead assertion */
+        function verifyForLookahead(
+            regexpContext: RegExpContext,
+            assertion: LookaheadAssertion,
+        ) {
+            for (const alternative of assertion.alternatives) {
+                const last =
+                    alternative.elements[alternative.elements.length - 1]
+                if (
+                    last?.type !== "Assertion" ||
+                    last.kind !== "lookahead" ||
+                    last.negate
+                ) {
+                    continue
+                }
+                // The last lookahead positive assertion within
+                // a lookahead assertion is the same without the assertion.
+                reportLookaroundAssertion(regexpContext, last)
+            }
+        }
+
+        /** Verify for lookbehind assertion */
+        function verifyForLookbehind(
+            regexpContext: RegExpContext,
+            assertion: LookbehindAssertion,
+        ) {
+            for (const alternative of assertion.alternatives) {
+                const first = alternative.elements[0]
+                if (
+                    first?.type !== "Assertion" ||
```

Comment:
While this does work correctly for empty alternatives, I don't like that TypeScript thinks that the type is `first: Element`. If I removed the `?`, TypeScript wouldn't be able to catch the bug.

The [`at` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at) is what would be ideal here, I think. But I don't think that we can use it yet because of browser/node support. Maybe we could make our own `at` method and use that?